mesos/match.go

```diff
@@ -68,6 +77,28 @@ func (m *attributeMatcher) Matches(o interface{}) error {
 	return errors.New("")
 }
 
+func (m *availabilityMatcher) Matches(o interface{}) error {
+	offer := o.(*mesosproto.Offer)
+
+	if offer.Unavailability == nil {
+		return nil
+	}
+
+	now := time.Now().UnixNano()
```

Comment:
A bit of a nitpick but timestamps always puts my in worry about everything mode. So I think getting the current timestamp is better to do when we create the matcher (sent as an argument even) so the implementation can remain completely pure.